source/tutorials/ad-hoc-developer-environments.rst

```diff
@@ -0,0 +1,137 @@
+Ad hoc developer environments
+=============================
+
+Assuming you have :ref:`Nix installed <install-nix>`, it is able to download packages
+and provide a new shell with packages available.
+
+This is a great way to play with Nix tooling and see some of its potential.
+
+
+When are such environments useful?
```

Comment:
Before answering _“when are such environments useful?”_ you should probably answer _“what is a shell environment?”_ so that the reader can more easily objectivize this idea. People have an easier time understanding things they can point a finger at.

It's fine if the answers to these to questions overlap a bit, it'll only reinforce the idea.

Reply:
I recommend a snippet similar to the following showing the general idea.

```
$ hello             
The program ‘hello’ is currently not installed.

$ nix-shell -p hello             

[nix-shell:~]$ hello
Hello, world!

[nix-shell:~]$ exit
exit

$ hello             
The program ‘hello’ is currently not installed.
```

Reply:
Great tip!